Optimize the pipeline by reduce the disk usage and build time

We are starting to hit the 'not enough space' issue(see  https://dev.azure.com/ms/react-native-windows/_build/results?buildId=47129). Currently I workaround the problem by deleting the buildout of ReactWindows-Desktop.sln before it builds Playground project.

There are possible two places we can improve:
1. Reduce the build time of some jobs
2. We didn't reuse the buildout put of vnext/ReactWindows-Desktop.sln in the next Playground project. so it hits `not enough space`.

not enough space error:
```
 ##[error]C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\amd64\Microsoft.Common.CurrentVersion.targets(4365,5): Error MSB3021: Unable to copy file "d:\a\1\s\vnext\target\x64\Debug\ReactUWP\React.UWP.pdb" to "d:\a\1\s\packages\playground\windows\x64\Debug\Playground\React.UWP.pdb". There is not enough space on the disk.
```

--- a/change/react-native-windows-2019-12-13-11-33-23-PCHDelete.json
+++ b/change/react-native-windows-2019-12-13-11-33-23-PCHDelete.json
@@ -1,0 +1,8 @@
+{
+  "type": "none",
+  "comment": "Delete .PCH ",
+  "packageName": "react-native-windows",
+  "email": "licanhua@live.com",
+  "commit": "42a383343b2b7624fb05efd5253064d9102746de",
+  "date": "2019-12-13T19:33:23.242Z"
+}

--- a/vnext/Directory.Build.targets
+++ b/vnext/Directory.Build.targets
@@ -47,4 +47,26 @@
     <Message Text="=> VCTargetsPath           [$(VCTargetsPath)]" />
   </Target>
 
+  <Target Name="_ComputePrecompToCleanUp">
+    <ItemGroup>
+      <PCHFileToClean Include="$(IntDir)\**\*.pch" />
+      <_PCHFileToCleanWithTimestamp Include="@(PCHFileToClean)" Condition="'%(Identity)' != ''">
+        <LastWriteTime>$([System.IO.File]::GetLastWriteTime('%(Identity)'))</LastWriteTime>
+      </_PCHFileToCleanWithTimestamp>
+    </ItemGroup>
+  </Target>
+  <!-- 
+    As a cppwinrt project, .pch is very big, and delete it in pipeline after build
+    Instead of outright deleting the PCHs, we want to trick the "project freshness detector" by
+    emitting empty files that look suspiciously like the PCHs it's expecting.
+    It's of utmost importance that their timestamps match up. -->
+  <Target Name="CleanUpPrecompHeaders"
+          DependsOnTargets="_ComputePrecompToCleanUp"
+          AfterTargets="AfterBuild"
+          Condition="'$(AGENT_ID)' != ''">
+    <!-- We just need to keep *ReactNativeWindows*'s PCHs because they get rebuilt more often. -->
+    <Delete Files="@(_PCHFileToCleanWithTimestamp)"/>
+    <Touch Files="@(_PCHFileToCleanWithTimestamp)" Time="%(LastWriteTime)" AlwaysCreate="true" />
+    <Message Text="PCH and Precomp object @(_PCHFileToCleanWithTimestamp) has been deleted for $(ProjectName)." />
+  </Target>
 </Project>

